### PR TITLE
Fix perks not shown in roundend and also perks sprites and hud.

### PR DESCRIPTION
**Pull request by Xackii**
## About The Pull Request

Fixes that perks not shown in wizard purchases in roundend report.
Added perks sprites: they show up when you buy them.
Снимок экрана 2024-10-18 161649
Ingame:
Снимок экрана 2024-10-18 195713

## Why It's Good For The Game

Fixes is good. 
Sprites to understand what perks you bought.

## Changelog

:cl:
qol: Purchased perks now shown in hud.
fix: Perks purchased by a wizard now shown in roundend report.
/:cl:

### CODE REVIEW

File: code/_onclick/hud/perks.dm [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+/atom/movable/screen/perk/fourhands
+	name = "four hands"
+	desc = /datum/spellbook_entry/perks/fourhands::desc
+	icon_state = "fourhands"
+
+/atom/movable/screen/perk/magnetism
+	name = "magnetism"
+	desc = /datum/spellbook_entry/perks/magnetism::desc
+	icon_state = "magnetism"
```
Comment (SmArtKar) [inline code/_onclick/hud/perks.dm:48]: These should definitely not be premade, and instead be a single type and have their name/desc/icon assigned when created in create_hud_icon

**Review by SmArtKar — DISMISSED** [review]

File: code/_onclick/hud/perks.dm (hunk 1 above)
Comment (Xackii) [reply code/_onclick/hud/perks.dm:48]: They can be variables? For some reason I decided that hud things should be ready-made templates and added when needed.

**Comment by Xackii** [comment]
We need craftable wands

**Review by Ghommie — CHANGES_REQUESTED** [review]
It looks nice, but I wish there's an action to hide these perks for less visual clutter.

**Review by Wallemations — APPROVED** [review]
Please make the perk icons togglable, but good work

**Comment by Xackii** [comment]
2024-10-2607-34-33-ezgif com-crop

**Comment by Xackii** [comment]
Was a bit busy

**Review by Wallemations — APPROVED** [review]
I ain't a code maint but here's a double-approval for the toggle addition

File: code/_onclick/hud/perks.dm [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	if(!isliving(usr))
+		return
+	var/mob/living/usr_is_living = usr
+	var/datum/antagonist/wizard/wizard_datum = usr_is_living.mind.has_antag_datum(/datum/antagonist/wizard)
+	if(!wizard_datum)
+		return
+	compact = !compact
+	var/datum/hud/user_hud = usr_is_living.hud_used
+	if(compact)
+		for(var/atom/movable/screen/perk/perk_on_hud in perks_on_hud)
```
Comment (Ghommie) [inline code/_onclick/hud/perks.dm:27]:
```suggestion
		for(var/atom/movable/screen/perk/perk_on_hud as anything in perks_on_hud)
```

File: code/_onclick/hud/perks.dm [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+			perks_compacted += perk_on_hud
+			user_hud.infodisplay -= perk_on_hud
+		user_hud.show_hud(user_hud.hud_version)
+	else
+		for(var/atom/movable/screen/perk/perk_compacted in perks_compacted)
```
Comment (Ghommie) [inline code/_onclick/hud/perks.dm:32]:
```suggestion
		for(var/atom/movable/screen/perk/perk_compacted as anything in perks_compacted)
```

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Review by Ghommie — APPROVED** [review]

**Comment by Ghommie** [comment]
I was a bit busy too.

**Pull request merged**